src/server/internalServer.cpp

```diff
@@ -306,6 +316,10 @@ SearchInfo InternalServer::getSearchInfo(const RequestContext& request) const
 {
   auto bookIds = selectBooks(request);
   checkBookNumber(bookIds.second, m_multizimSearchLimit);
+  if ( getLanguages(*mp_library, bookIds.second).size() != 1 ) {
```

Comment:
I was not certain that this would be better but implemented it anyway. See the fixup commit. It ended up even worse than I expected it to be. I think you will agree that we better not include that change in the PR.

Reply:
> I think you will agree that we better not include that change in the PR.

I don't see why. Can you explain why you don't want ot include this change ?
You indeed have to pass the set of bookId to inner functions but there is not a lot we can do.
I would have add an overloading method (`std::vector<std::string> getBooksLanguages(const BookIdCollection& bookIds) const` and `std::vector<std::string> getBooksLanguages() const`) instead of a default argument but it would not fundamentally change your implementation.


---

Another solution would be to add another intermediate class like `BookSet` and implement some helper methods on it (`getLanguages`, `getPublisher`, ...) and have the `Library` inherit on `BookSet` and have a `BookSet getBookSet(std::set<std::string> bookIds)`. (And we could have `OPDSDumper` also working on a `BookSet`).
But if you go this way, it should be in another PR.

Reply:
> I don't see why. Can you explain why you don't want ot include this change ?

Because it is too extensive for the small issue it is intended to address. Also in its current form it makes the API slightly inconsistent. An optional `bookIds` parameter was added only to `Library::getBooksLanguages()` and `Library::getBooksLanguagesWithCounts()` (the enhanced version of the latter not being justified by any current usage). From the abstract API's point of view it is not clear why `Library::getBooksCategories()`, `Library::getBooksCreators()` and `Library::getBooksPublishers()` don't deserve the same enhancement (in which case the change would become even larger).

> You indeed have to pass the set of bookId to inner functions but there is not a lot we can do.

For the time being we can just get along with a custom small & simple helper function which keeps the total complexity of the full code at a lower level. Enhancing API's on every opportunity is not a good strategy. The ultimate goal should be maintaining globally optimal complexity.

Reply:
Yes, let move on with your small helper. If we need API enhancement in the future, we will make it.

Reply:
I undid/dropped the fixup commit made in response to this review comment.